contracts/feeManager/FeeManagerLogicV1.sol

```diff
@@ -0,0 +1,285 @@
+        }
+
+        uint256 remainingFunds = _payBeneficiaries(debtor, msg.value);
+
+        if (remainingFunds > 0) {
+            remainingFunds = _payOwner(debtor, remainingFunds);
+
+            if (remainingFunds > 0) {
+                _funds[msg.sender] += remainingFunds;
+            }
+        }
+    }
+
+    /**
+     * @notice Pays fees from the debtor to beneficiaries
+     * @param debtor The address of the debtor
+     * @param funds The funds available to pay beneficiaries
+     * @return The remaining funds after paying the beneficiaries
+     */
+    function _payBeneficiaries(address debtor, uint256 funds)
+        internal
+        returns (uint256)
+    {
+        while (funds > 0 && _creditors[debtor].length > 0) {
+            address beneficiary = _creditors[debtor][0];
+            bytes32 amountsKey = keccak256(
+                abi.encodePacked(debtor, beneficiary)
+            );
+
+            uint256 payment = 0;
+            if (_amounts[amountsKey] > funds) {
+                payment = funds;
+            } else {
+                payment = _amounts[amountsKey];
+                _creditors[debtor][0] = _creditors[debtor][
+                    _creditors[debtor].length - 1
+                ];
+                _creditors[debtor].pop();
+            }
+
+            funds -= payment;
+            _amounts[amountsKey] -= payment;
+            _funds[beneficiary] += payment;
+
+            emit FeePayment(debtor, beneficiary, payment);
+        }
+
+        return funds;
+    }
+
+    /**
+     * @notice Pays fees from the debtor to the gateway
+     * @param debtor The address of the debtor
+     * @param funds The funds available to pay the gateway
+     * @return The remaining funds after paying the gateway
+     */
+    function _payOwner(address debtor, uint256 funds)
+        internal
+        returns (uint256)
+    {
+        bytes32 amountsKey = keccak256(abi.encodePacked(debtor, _feesOwner));
+        uint256 ownerPayment = _amounts[amountsKey] < funds
+            ? _amounts[amountsKey]
+            : funds;
+
+        funds -= ownerPayment;
+        _amounts[amountsKey] -= ownerPayment;
+        _funds[_feesOwner] += ownerPayment;
+
+        emit FeePayment(debtor, _feesOwner, ownerPayment);
+
+        return funds;
+    }
+
+    /**
+     * @notice Overrides the transferOwnership function of Ownable
+     * @param newOwner The address of the new owner
+     */
+    function transferOwnership(address newOwner)
+        public
+        override(IOwnable, InitializableOwnable)
+    {
+        if (newOwner == owner()) revert NewOwnerIsCurrentOwner();
+        super.transferOwnership(newOwner);
+    }
+
+    /**
+     * @inheritdoc IFeeManager
+     */
+    function getGatewayFeesOwner() public view override returns (address) {
+        return _feesOwner;
+    }
+
+    /**
+     * @inheritdoc IFeeManager
+     */
+    function setRIFGateway(IRIFGateway rifGateway) public override {
```

Comment:
I believe this function should have `onlyOwner` modifier or some other form of authorization. Otherwise, attackers can call this function and set the `rifGateway` address to arbitrary address.

Reply:
I solve this issue on [#PR89](https://github.com/rsksmart/rif-everyday-gateway/pull/89/files#diff-df1111fb6b479c1a50667ee6835529dd95ddb8e809294a8deb70aa969a26f45f).